src/main/java/au/com/rayh/Keychain.java

```diff
@@ -1,18 +1,21 @@
 package au.com.rayh;
 
+import hudson.util.Secret;
+import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 public class Keychain {
     private String keychainName;
     private String keychainPath;
-    private String keychainPassword;
+    private Secret keychainPassword;
     private Boolean inSearchPath;
 
     public Keychain() {
     }
 
     @DataBoundConstructor
-    public Keychain(String keychainName, String keychainPath, String keychainPassword, Boolean inSearchPath) {
+    public Keychain(String keychainName, String keychainPath, Secret keychainPassword, Boolean inSearchPath) {
```

Comment:
Is it not better to add an other default constructor like "public Keychain()" to use @DataBoundSetter? Like this you can deprecate this one and modify the setting of keychainPassword adding Secret.fromString().